src/features/game/state/GameStateManager.ts

```diff
@@ -454,6 +458,55 @@ class GameStateManager {
     return this.checkpointTask.getAllVisibleTaskData();
   }
 
+  ///////////////////////////////
+  //          Quiz             //
+  ///////////////////////////////
+
+  /**
+   * Checks whether a quiz has been obtained full marks.
+   *
+   * @param key quiz id
+   * @returns {boolean}
+   */
+  public isQuizComplete(quizId: string): boolean {
+    return this.quizScores.get(quizId) === GameGlobalAPI.getInstance().getQuizLength(quizId);
+  }
+
+  /**
+   * Checks whether a specific quiz has been played.
+   *
+   * @param key quiz id
+   * @returns {boolean}
+   */
+  public isQuizAttempted(quizId: string): boolean {
+    return this.quizScores.has(quizId);
+  }
+
+  /**
+   * Get the score of a quiz.
+   * Return 0 if the quiz has not been played.
+   *
+   * @param quizId
+   * @returns
+   */
+  public getQuizScore(quizId: ItemId): number {
+    const score = this.quizScores.get(quizId);
+    if (score) {
+      return score;
+    }
+    return 0;
+  }
+
+  /**
+   * Set the score of a quiz to a given number.
+   *
+   * @param quizId The id of the quiz.
+   * @param newScore The new score to be set.
+   */
+  public setQuizScore(quizId: string, newScore: number) {
```

Comment:
Ditto on typing.

Reply:
used in GameGlobalAPI and GameQuizManager

Reply:
Resolving the conversation for now as changing access is needed for `GameGlobalAPI`. (GameGlobalAPI is a separate issue that is out of scope for this PR 😓)